paddle/gserver/layers/TransLayer.cpp

```diff
@@ -56,7 +56,14 @@ void TransLayer::backward(const UpdateCallback& callback) {
     return;
   }
   MatrixPtr preGrad = getInputGrad(0);
-  outputGrad->transpose(preGrad, false);
+  if (preGrad) {
+    MatrixPtr transGrad = Matrix::create(preGrad->getHeight(),
```

Comment:
Does not need to create a memory outside, you may use transpose(transGrad, true) to allocate the memory inside